dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/Standard/ACVoltageControl/QRefLim.mo

```diff
@@ -0,0 +1,88 @@
+within Dynawo.Electrical.HVDC.Standard.ACVoltageControl;
+
+/*
+* Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model QRefLim
+
+  import Modelica;
+  import Dynawo;
+  import Dynawo.Types;
+  import Dynawo.Connectors;
+  import Dynawo.Electrical.SystemBase;
+
+  extends Parameters.Params_QRefLim;
+
+  Modelica.Blocks.Interfaces.RealInput QRefUQPu(start = Q0Pu) "Reference reactive power in U mode in p.u (base SNom)" annotation(
+    Placement(visible = true, transformation(origin = {-120, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-110, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput PPu(start = P0Pu) "Active power in p.u (base SNom)" annotation(
+    Placement(visible = true, transformation(origin = {-120, 75}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-110, -80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput UPu(start = U0Pu) "Voltage in p.u (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {-120, -75}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-110, 80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  Modelica.Blocks.Interfaces.RealOutput QRefLimPu(start = Q0Pu) "Reference reactive power in p.u (base SNom) after applying the diagrams" annotation(
+    Placement(visible = true, transformation(origin = {110, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  Modelica.Blocks.Nonlinear.Limiter limiter(limitsAtInit = true, uMax = QMaxOPPu, uMin = QMinOPPu)  annotation(
+    Placement(visible = true, transformation(origin = {-41, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Nonlinear.VariableLimiter variableLimiter annotation(
+    Placement(visible = true, transformation(origin = {-4, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Nonlinear.VariableLimiter variableLimiter1 annotation(
+    Placement(visible = true, transformation(origin = {30, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Tables.CombiTable1D QMaxPPu(extrapolation = Modelica.Blocks.Types.Extrapolation.HoldLastPoint, table = tableQMaxPPu) annotation(
```

Comment:
extrapolation doesn't work with my OM. Does it work for you @adriguir @RomainLosseauRTE

Reply:
in OM sorry

Reply:
I believe it works. How did you notice that it didn't work? Perhaps it doesn't work but I haven't paid attention...

Reply:
I couldn't compile or simulate... I get the following error:
[2] 14:30:40 Traduction Erreur
In modifier (extrapolation = Modelica.Blocks.Types.Extrapolation.HoldLastPoint), class or component extrapolation not found in <Modelica.Blocks.Tables.CombiTable1D$HVDC$UdcPu_Side$aCVoltageControl$qRefLim$QMaxPPu>.

Reply:
Oh, this really seems to come from different OM or Modelica versions or set-ups. It simulates without problem into my OM. Does the type exist into your Modelica Standard Library? I mean HoldLastPoint into Modelica.Blocks.Types.Extrapolation.

Reply:
Yes.. I have MSL 3.2.2. If I remove it I can simulate and it looks like I get the right results so should we keep it (it would be less dependent on the version...) @cossartque ??

Reply:
Well, given that we have another point creating a horizontal line, I think we can remove it.